File: armstrong.py
def checkarmstrong(num):
    a=num
    y=len(str(num))
    sum_of_number=0
    while (num>0):
        sum_of_number+=(num%10)**y
        num=num//10
    print(sum_of_number)
    if a==sum_of_number:
        print(a,"is a armstrong")
    else:
        print(a,"is not an armstrong number")
checkarmstrong(153)
checkarmstrong(1634)
checkarmstrong(1758)

File: armstrong.c

#include <math.h>
#include <stdio.h>

void checkarmstrong(int num);

int main()
{
  int a;
  printf("Enter any any number:\n");
  scanf("%d", &a);
  checkarmstrong(a);
  return 0;
}

void checkarmstrong(int num)
{
  int sum = 0;
  int temp = num;
  int power = (int)log10(num) + 1;
  while (num > 0)
  {
    sum += pow(num % 10, power);
    num = num / 10;
  }
  (sum == temp) ? printf("%d is an armstrong number.\n", temp) : printf("%d is not an armstrong number\n", temp);
}
